### PR TITLE
[fix] Ensure we check deepspeed/sharded in multinode DDP

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Fix a user bug where when sharded is selected, we hard crash saying to select a support distributed type. Sharded is a valid option, modified the check to use properties defined for future proofing.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by SeanNaren** [comment]
Do not merge, tests are failing (not sure why you can even merge)

File: tests/accelerators/test_accelerator_connector.py [hunk 1]
```diff
@@ -415,3 +417,20 @@ def test_plugin_accelerator_choice(accelerator, plugin):
 
     trainer = Trainer(plugins=plugin, num_processes=2)
     assert isinstance(trainer.accelerator.training_type_plugin, DDPShardedPlugin)
+
+
+@pytest.mark.parametrize(
+    ["accelerator", "plugin"],
+    [('ddp', DDPPlugin), ('ddp_spawn', DDPSpawnPlugin), ('ddp_sharded', DDPShardedPlugin),
+     ('ddp_sharded_spawn', DDPSpawnShardedPlugin), ('deepspeed', DeepSpeedPlugin)],
+)
```
Comment (awaelchli) [inline tests/accelerators/test_accelerator_connector.py:427]: Looks like all this needs is a skip condition if deepspeed is not available. 
You could do that with 

`pytest.param("deepspeed", DeepSpeedPlugin, marks=pytest.mark.xfail(reason="needs deepspeed"))`

I believe

**Review by carmocca — APPROVED** [review]

File: tests/accelerators/test_accelerator_connector.py (hunk 1 above)
Comment (SeanNaren) [reply tests/accelerators/test_accelerator_connector.py]: omg thank you! let me add the mocks back
